dxr/plugins/python/indexers.py

```diff
@@ -295,14 +302,23 @@ def get_node_start_end(self, node):
         """
         loc = node.lineno, node.col_offset
 
-        if isinstance(node, ast.ClassDef) or isinstance(node, ast.FunctionDef):
-            start, end = self.node_start_table.get(loc, (None, None))
-        elif isinstance(node, ast.Call):
-            start, end = self.call_start_table.get(loc, (None, None))
-        else:
-            start, end = None, None
+        if isinstance(node, (ast.ClassDef, ast.FunctionDef, ast.Call)):
+            loc_list = self.node_start_table.get(loc, [])
+            if loc_list:
+                return loc_list[-1]
+
+        return None, None
 
-        return start, end
+    def advance_node(self, node):
+        """Destructively change which actual token offset we'll get on a call
+        to get_node_start_end.
+
+        """
+        loc = node.lineno, node.col_offset
+        try:
+            self.node_start_table[loc].pop()
+        except Exception:
```

Comment:
We should avoid except Exception, presumably it's one or both of KeyError and IndexError.